src/main/scala/nl/knaw/dans/lib/encode/package.scala

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (C) 2016 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.lib
+
+import com.google.common.net.PercentEscaper
+import java.nio.file.Path
+import scala.collection.JavaConverters._
+
+package object encode {
+
+  // PercentEscaper that escapes all characters except '_' and English alphanumerical characters
+  private val bagStorePercentEscaper = new PercentEscaper("_", false)
+
+  implicit class PathEncoding(val path: Path) extends AnyVal {
+
+    /**
+     * Escapes all characters in path segments, except those declared as "safe characters"
+     *
+     * @example
+     * {{{
+     *    import nl.knaw.dans.lib.encode.PathEncoding
+     *    import java.nio.file.Paths
+     *
+     *    val path = Paths.get("an/example/path-123/file.txt)"
+     *    val escapedPath = path.escapePath
```

Comment:
1. could you replace the current example with one that has more characters that are being escaped
2. please also provide an output for this example (in a comment below this line)

same for the `String` extension below, please.

Reply:
done